share/eds/byzantine/bad_encoding.go

```diff
@@ -184,19 +177,25 @@ func (p *BadEncodingProof) Validate(hdr *header.ExtendedHeader) error {
 	}
 
 	// verify that Merkle proofs correspond to particular shares.
-	shares := make([][]byte, len(merkleRoots))
+	shares := make([][]byte, width)
 	for index, shr := range p.Shares {
 		if shr == nil {
 			continue
 		}
 		// validate inclusion of the share into one of the DAHeader roots
-		if ok := shr.Validate(ipld.MustCidFromNamespacedSha256(merkleRoots[index])); !ok {
+		var x, y int
+		if p.Axis == rsmt2d.Row {
+			x, y = index, int(p.Index)
+		} else {
+			x, y = int(p.Index), index
+		}
+		root := getRoot(hdr.DAH, shr.Axis, x, y)
+		rootCid := ipld.MustCidFromNamespacedSha256(root)
+		if ok := shr.Validate(rootCid, x, y, width); !ok {
```

Comment:
I propose to pass the DAH to Validate and do this logic there instead.

Reply:
Coordintes are defined by befp axis, not by proof axis. Those are not relateed. Refactored whole Validate thing to work over axisType,axisIdx,shrIdx instead of using x,y. Let me know how you like it